common/submit.py

```diff
@@ -49,11 +54,13 @@ def store_submit(request: HttpRequest, assignment: AssignedTask) -> Submit:
     submits = Submit.objects.filter(assignment__pk=assignment.id, student__pk=request.user.id)
 
     # Check submit date across all tasks, just to be a bit more defensive
-    last_submit_date = Submit.objects.filter(student=request.user).order_by("-created_at").first()
-    if last_submit_date is not None:
-        since_last_submit = (
-            datetime.datetime.now(datetime.timezone.utc) - last_submit_date.created_at
-        )
+    last_submit = Submit.objects.filter(student=request.user).order_by("-created_at").first()
+
+    if last_submit is not None:
+        if assignment.task.type == "exam" and last_submit.is_final:
```

Comment:
Let's check only `is_final` and not the task type, to avoid scattering the logic for what does a final submit mean across the codebase. We will make sure elsewhere that only exam submits can be marked as final (for now).